translate/src/hooks/useNarrowScreen.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * Return true if the screen is narrower than 600 px. Useful in Responsive Web Design.
+ */
+export function useNarrowScreen(): boolean {
+  const [width, setWidth] = useState(window.innerWidth);
+
+  useEffect(() => {
+    const handleWindowResize = () => setWidth(window.innerWidth);
```

Comment:
This state change is going to trigger a re-render of the whole component. The state variable should be the boolean result of the comparison to avoid that.